### DISCUSSION TITLE
Documentation contributions

### DISCUSSION THREAD

**Discussion by vriesdemichael** [ggml-org/llama.cpp#5683] [General]
Hey there,

I am trying to understand various concepts implemented in this repository, but find myself crawling through c++ code for most of the answers.

I get that llama.cpp has been rapidly growing, and that documentation is not the highest priority. 
While trying to understand the codebase, I will gladly add documentation where it is missing.

I see several attempted strategies for documentation:
- code documentation in the header and cpp files.
- readme.md in most of the examples
- a `docs` folder in the repository root (which seems mostly unused)

When trying to add more documentation, which place should we focus on? A little guidance would be helpful.

**Comment by ggerganov** [comment]
The readme files of the examples can be improved. At the very least, adding sample usage commands (ones that you can just copy-paste in the terminal) and expected results would be very helpful.

Documenting the code is mostly done in the headers / APIs, but it's indeed rudimentary at this time.
Maybe adding more code comments to the source of the examples would be helpful.
